src/urllib3/connection.py

```diff
@@ -488,17 +489,18 @@ def connect(self) -> None:
                 ssl_maximum_version=self.ssl_maximum_version,
                 cert_reqs=resolve_cert_reqs(self.cert_reqs),
             )
-            # In some cases, we want to verify hostnames ourselves
-            if (
-                # `ssl` can't verify fingerprints or alternate hostnames
-                self.assert_fingerprint
-                or self.assert_hostname
-                # We still support OpenSSL 1.0.2, which prevents us from verifying
-                # hostnames easily: https://github.com/pyca/pyopenssl/pull/933
-                or ssl_.IS_PYOPENSSL
-                or not ssl_.HAS_NEVER_CHECK_COMMON_NAME
-            ):
-                self.ssl_context.check_hostname = False
+
+        # In some cases, we want to verify hostnames ourselves
```

Comment:
I don't think we want to modify every `ssl_context` we get passed, only the default one that we create ourselves?